### PR TITLE
modified agmip output script to use piaminterfaces

### PR DESCRIPTION
**Pull request by jdrtommey**
## :bird: Description of this PR :bird:

- This PR modifies the agmip output script to use the piamInterfaces mapping instead of the deprecated mapping write.reportProject from the iamc package. 

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts
  - **Medium risk**: Uncritical changes in the model core (e.g. moderate modifications in non-default realizations)
  - **High risk**: Critical changes in model core or default settings (e.g. changing a model default or adjusting a core mechanic in the model)

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
  - **Medium risk**: 
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
  - **High risk**:
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
    - Default run from the PR target branch for comparison
    - Provide relevant comparison plots (land-use, emissions, food prices, land-use intensity,...)

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - Current develop branch default : ** mins
  - This PR's default :  ** mins

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [ ] content review done (at least 1)
- [ ] RSE review done (at least 1)

### CODE REVIEW

File: scripts/output/projects/agmip_merge_report.R [hunk 1]
```diff
@@ -64,10 +65,32 @@ if (!is.null(missing)) {
...
-  write.reportProject(mif = "output/agmip_report_full.csv",
-                      mapping = system.file("extdata", mapping = "variablemappingAgMIP.csv", package = "magpie4"),
-                      file = "output/agmip_report_subset.csv", format = "AgMIP")
-  #write.reportProject(mif="output/agmip_report_full.csv",mapping = "mapping_magpie_agmip.csv", file = "agmip_report_subset.csv",format="AgMIP")
+  submission <- generateIIASASubmission(
+    mifs = "output/agmip_report_full.csv",
+    mapping = "AgMIP",
+    model = "MAgPIE",
+    outputFilename = NULL,
+    timesteps = c(seq(1900, 2100, 1)),
```
Comment (FelicitasBeier) [inline scripts/output/projects/agmip_merge_report.R:73]: Why are the time steps from 1900 onwards? Does that work? I think MAgPIE only provides output starting from 1995.

**Review by FelicitasBeier — COMMENTED** [review]
Thank you for adding this! 

Just one question: Wasn't there some special treatment (in terms of units or something) required in this script because we couldn't introduce the special rules in piamInterfaces? 

And small comment: it's good practice to fill out the first tab "Conversation" "Explain the purpose of this PR" such that it's clear what this PR is for.

File: scripts/output/projects/agmip_merge_report.R (hunk 1 above)
Comment (jdrtommey) [reply scripts/output/projects/agmip_merge_report.R:73]: Yea I should change this to the actual range we expect, this was just to make sure nothing got filtered out when testing and never changed it, I will set to 1995 to 2100

Comment (jdrtommey) [reply scripts/output/projects/agmip_merge_report.R:73]: also just overlooked adding a description, my bad! have added a brief one now. 

And on the PR for the piamInterfaces PR agreed that exceptions to the mappings should really be best handled in one place i.e, In piamInterfaces. So we have left the inconsistent units in there for now and will add additional flexibility into piamInterfaces to allow for exceptions to be added on a per mapping basis, so the /year issues in AgMIP do not effect the other mappings.

**Review by FelicitasBeier — APPROVED** [review]

**Review by tscheypidi — APPROVED** [review]

**Pull request merged**